File: rps.rb
CHOICES = {r: 'rock', p: 'paper', s: 'scissors'}

def design
  puts "***********************"
end

puts "What is your name player?"
user_name = gets.chomp.capitalize

begin
  design
  puts "#{user_name}! Choose rock, paper, or scissors to begin game."
  design
  player_input = gets.chomp

  computer_choice = CHOICES.values.sample

  if player_input == computer_choice
    puts "#{user_name} choose #{player_input} and Computer choose #{computer_choice}." 
    puts "Its a Tie!"
    design
  elsif player_input == 'rock' && computer_choice == 'scissors' || 
    player_input == 'paper' && computer_choice == 'rock' || 
    player_input == 'scissors' && computer_choice == 'paper'
    puts "#{user_name} choose #{player_input} and Computer choose #{computer_choice}."
    puts "Y O U   W I N  !  !  !"
    design
  else
    puts "#{user_name} choose #{player_input} and Computer choose #{computer_choice}"
    puts "Y O U   L O S E  !  !  !"
    design
  end
  puts "Play again (y/n)"
  done = gets.chomp
  system("clear")
end while done == "y"

File: rps.py
from random import choice
import os

def clear():
    os.system('clear')
 
choices = ['rock', 'paper', 'scissors']
 
def design():
	print('*********************************')
 
def winning_message(player_choice, computer_choice):
	player_score = 0
	print("Player choose %s"%(player_choice))
	print("Computer choose %s"%(computer_choice))
	print('you win')
	design()
	player_score += 10
	return player_score
 
computer_score = 0
player_score = 0
play_again = 'y'
 
while play_again == 'y':
 
	design()
	print("choose rock(r), paper(p) or scissors(s) to begin.")
	design()
	player_choice = raw_input(">>> ").lower()
	computer_choice = choice(choices)	
	design()
	
	if player_choice[0] == computer_choice[0]:
		print("Player choose %s"%(player_choice))
		print("Computer choose %s"%(computer_choice))
		print('its a tie!')
		design()
	elif player_choice[0] == 'r' and computer_choice[0] == 's':
		player_score += winning_message(player_choice, computer_choice)
	elif player_choice[0] == 'p' and computer_choice[0] == 'r':
		player_score += winning_message(player_choice, computer_choice)
	elif player_choice[0] == 's' and computer_choice[0] == 'p':
		player_score += winning_message(player_choice, computer_choice)
	else:
		print("Player choose %s"%(player_choice))
		print("Computer choose %s"%(computer_choice))
		print('you lose')
		design()
		computer_score += 10
		
	print('Play Again? [y/n]')
	play_again = raw_input('>>>').lower()
	clear()
	print("Player: %s"%(player_score))
	print("Computer: %s"%(computer_score))
 



